backend/embedding/main.py

```diff
@@ -169,44 +190,77 @@ def main(
             return
 
         # Calculate embeddings using LangChain
-        contents: list[str] = []
-        sources: list[str] = []
-        embeddings: list[list[float]] = []
-
-        if len(source_urls) > 0:
-            embed(
-                UrlLoader(source_urls),
-                contents,
-                sources,
-                embeddings,
-                chunk_size,
-                chunk_overlap,
-            )
-        if len(sitemap_urls) > 0:
-            for sitemap_url in sitemap_urls:
-                raise NotImplementedError()
-        if len(filenames) > 0:
-            for filename in filenames:
+        with multiprocessing.Manager() as manager:
```

Comment:
Have you checked that `source_urls` also works well? (The embedding task for url is not the `future` task).

Reply:
(In the near future, we also might want to refactor the `UrlLoader` so that the urls processings also can be parallelized)

Reply:
insert_to_postgres doesn't work when source_urls is running.
There is an error in Python indentation.
I will fix it.

Reply:
Python indentation error was fixed